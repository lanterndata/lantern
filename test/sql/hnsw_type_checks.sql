CREATE EXTENSION IF NOT EXISTS lanterndb;

\set ON_ERROR_STOP off
DROP TABLE IF EXISTS small_world;
CREATE TABLE small_world (
    id varchar(3),
    vector real[]
);

INSERT INTO small_world (id, vector) VALUES ('001', '{0,0,0,0}');

CREATE INDEX ON small_world USING hnsw (vector) WITH (dims=4);

-- should throw dimension error
INSERT INTO small_world (id, vector) VALUES ('002', '{0,0,0}');

-- should insert successfully
INSERT INTO small_world (id, vector) VALUES ('003', '{0,0,0,0}');
