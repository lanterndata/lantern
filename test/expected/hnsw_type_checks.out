CREATE EXTENSION IF NOT EXISTS lanterndb;
NOTICE:  extension "lanterndb" already exists, skipping
\set ON_ERROR_STOP off
DROP TABLE IF EXISTS small_world;
CREATE TABLE small_world (
    id varchar(3),
    vector real[]
);
INSERT INTO small_world (id, vector) VALUES ('001', '{0,0,0,0}');
CREATE INDEX ON small_world USING hnsw (vector) WITH (dims=4);
INFO:  done init usearch index
INFO:  inserted 1 elements
INFO:  done saving 1 vectors
-- should throw dimension error
INSERT INTO small_world (id, vector) VALUES ('002', '{0,0,0}');
ERROR:  Wrong number of dimensions: 3 instead of 4 expected
-- should insert successfully
INSERT INTO small_world (id, vector) VALUES ('003', '{0,0,0,0}');
